application/src/main/java/uk/nhs/hee/tis/revalidation/service/RecommendationElasticSearchService.java

```diff
@@ -50,4 +52,61 @@ public void addRecommendationViews(RecommendationView dataToSave) {
           dataToSave.getGmcReferenceNumber(), dataToSave.getTcsPersonId(), ex);
     }
   }
+
+  /**
+   * add new Recommendation trainee to elasticsearch index.
+   *
+   * @param dataToSave Recommendation trainee to go in elasticsearch
+   */
+  public void saveRecommendationViews(RecommendationView dataToSave) {
+    Iterable<RecommendationView> existingRecords = findRecommendationViewsByGmcNumber(
+        dataToSave.getGmcReferenceNumber());
+
+    // if doctor already exists in ES index, then update the existing record
+    if (Iterables.size(existingRecords) > 0) {
+      updateRecommendationViews(existingRecords, dataToSave);
+    }
+    // otherwise, add a new record
+    else {
+      addRecommendationViews(dataToSave);
+    }
+  }
+
+  /**
+   * update existing Recommendation to elasticsearch index.
+   *
+   * @param existingRecords existing Recommendation to be updated in elasticsearch
+   * @param dataToSave      new Recommendation details to be saved in elasticsearch
+   */
+  private void updateRecommendationViews(Iterable<RecommendationView> existingRecords,
+      RecommendationView dataToSave) {
+    existingRecords.forEach(recommendationView -> {
+      dataToSave.setId(recommendationView.getId());
+      try {
+        recommendationElasticSearchRepository.save(dataToSave);
+      } catch (Exception ex) {
+        LOG.info("Exception in `updateRecommendationViews` (GmcId: {}; PersonId: {}): {}",
+            dataToSave.getGmcReferenceNumber(), dataToSave.getTcsPersonId(), ex);
+      }
+    });
+  }
+
+  /**
+   * find iterable of RecommendationView from elasticsearch index.
+   *
+   * @param gmcReferenceNumber String to go in elasticsearch
+   */
+  private Iterable<RecommendationView> findRecommendationViewsByGmcNumber(
+      String gmcReferenceNumber) {
+    Iterable<RecommendationView> result = new ArrayList<>();
+
+    if (gmcReferenceNumber != null) {
```

Comment:
We probably want to throw an exception in the else clause, otherwise this will end up adding a record to recommendation index with a null gmc number